pkg/csi/service/vanilla/controller.go

```diff
@@ -251,6 +258,16 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 				}
 			}
 		}
+		if len(c.managers.VcenterConfigs) > 1 {
+			log.Info("Loading CnsVolumeInfo Service to persist mapping for VolumeID to vCenter")
```

Comment:
Even if there is only 1 VC, we should initialise volume info service.

This is because we will be deprecating single instance objects (applicable for a single VC setup) like metadataSyncer's volumeManager and host fields.

Instead even for a single VC setup, we will have a map having only 1 key value pair for host and volumeManager. We don't need to maintain multiple fields.

Reply:
The majority of deployment will be using a Single vCenter server. In my opinion, we should only be loading this service when we really need to keep a mapping of volumeID to VC map. 

Regarding metadata syncer changes, we do not want to make a lot of code changes for other flavors (WCP, Guest Cluster). Metadata syncer is shared among wcp, guest cluster and vanilla. Let's discuss how we can implement vanilla related changes for multi vcenter support without affecting WCP and Guest Cluster.